### PR TITLE
Closes the correct span on promise complete

### PR DESCRIPTION
**Pull request by adriancole**
Fixes a bad bug if you actually used the local tracing with a promise. I noticed found this using the react-native example. Immediate patch release to follow merge.

### CODE REVIEW

**Comment by adriancole** [comment]
going to merge and cut this as currently things are broke. welcome feedback after commit (this change is not api affecting, so safe to suggest rework!)

File: packages/zipkin/test/trace.test.js [hunk 1]
```diff
@@ -115,9 +115,7 @@ describe('Tracer', () => {
     ctxImpl.scoped(() => {
       const promise = trace.local('buy-smoothie', () =>
         Promise.delay(10)
-               .then(() => {
-                 throw new Error('no smoothies. try our cake');
-               })
+               .then(() => 'smoothie')
```
Comment (eirslett) [inline packages/zipkin/test/trace.test.js]: Shouldn't we have a unit test for the error case as well?

Comment (adriancole) [reply packages/zipkin/test/trace.test.js]: :) the red you are seein was me correcting code that did same check in 2 test cases! I fixed one to test success

Comment (adriancole) [reply packages/zipkin/test/trace.test.js]: Can add another test to verify the error calback sets correct trace ids though..

**Pull request merged**